tests/unittests/cmd/devel/test_logs.py

```diff
@@ -194,6 +209,77 @@ def fake_subp(cmd):
         )
         fake_stderr.write.assert_any_call("Wrote %s\n" % output_tarfile)
 
+    def test_write_command_output_to_file(self, m_getuid, tmpdir):
+        # what does this do???
+        m_getuid.return_value = 100
+        test_str = "cloud-init? more like cloud-innit!"
+        output_file1 = tmpdir.join("test-output-file-1.txt")
+        output_file2 = tmpdir.join("test-output-file-2.txt")
+        nonexistent_path = "/this-directory-does-not-exist"
+        expected_stderr = (
+            "Unexpected error while running command.\n"
+            + f"Command: ['ls', '{nonexistent_path}']\n"
+            + "Exit code: 2\n"
+            + "Reason: -\n"
+            + "Stdout: \n"
+            + f"Stderr: ls: cannot access '{nonexistent_path}': "
+            + "No such file or directory"
+        )
+        return_output1 = logs._write_command_output_to_file(
+            filename=output_file1,
+            cmd=["echo", test_str],
+            msg="",
+            verbosity=1,
+        )
+        return_output2 = logs._write_command_output_to_file(
+            filename=output_file2,
+            cmd=["ls", nonexistent_path],
+            msg="",
+            verbosity=1,
+        )
+
+        assert test_str + "\n" == return_output1
+        assert test_str + "\n" == load_file(output_file1)
+        assert expected_stderr == return_output2
+        assert expected_stderr == load_file(output_file2)
+
+    def test_stream_command_output_to_file(self, m_getuid, tmpdir):
```

Comment:
If this test is going to mostly be a duplicate of the one above it, it'd be a great candidate for [parametrization](https://docs.pytest.org/en/7.3.x/how-to/parametrize.html)